### PR TITLE
DAOS-2384 control: add format diagrams to readme

### PR DESCRIPTION
**Pull request by tanabarr**
Add description and diagrams explaining server and storage formatting
with client and server control plane applications.

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

### CODE REVIEW

**Comment by tanabarr** [comment]
> One item that isn't clear to me is the work flow with multiple storage nodes. The case of a single storage node with a single server and a single client application is clear. It would be helpful to show how storage format on multiple storage nodes is handled. For example, are we trying to show that daos_server responds to a single DMG format request for a single storage node at a time (and the admin must iterate through a list of storage devices manually)? Or does it get a single format request which translates into one format operation for each piece of unformatted storage distributed throughout the network (where the server looks at its config file to see what it knows about)?

these explanations should be added to the dmg readme, probably in a separate patch

File: src/control/server/README.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # DAOS Server (control-plane)
 
+## Workflow
+
+Control plane server (`daos_server`) instances will open and listen on gRPC channel for requests from control plane client applications. Administrators can perform provisioning operations on network and storage hardware through control plane [`dmg`](../dmg) management tool and when ready to start data plane, calling `dmg storage format` will enable the server to be formatted (superblock written) and start the data plane.
+
+![Server format diagram](/doc/graph/server_format_flow.png)
```
Comment (kjacque) [inline src/control/server/README.md]: Noticed a minor typo in the diagram: "Fotmat storage"

File: src/control/server/README.md [hunk 2]
```diff
@@ -1,5 +1,11 @@
 # DAOS Server (control-plane)
 
+## Workflow
+
+Control plane server (`daos_server`) instances will open and listen on gRPC channel for requests from control plane client applications. Administrators can perform provisioning operations on network and storage hardware through control plane [`dmg`](../dmg) management tool and when ready to start data plane, calling `dmg storage format` will enable the server to be formatted (superblock written) and start the data plane.
```
Comment (kjacque) [inline src/control/server/README.md]: Basic content looks good. A few minor suggestions for clarity:
"will open and listen on gRPC channel for requests" -> "will open a gRPC channel to listen for requests"
"through control plane (dmg) management tool" -> "through the control plane (dmg) management tool"
"and when ready to start data plane, ..." -> New sentence: "Calling `dmg storage format` formats the server, writes the superblock, and starts the data plane."

File: src/control/server/README.md (hunk 1 above)
Comment (tanabarr) [reply src/control/server/README.md]: annoying, I will redo

**Pull request merged**